openstates/new-openstates.org
# Add webpack build system

### Summary
This PR sets up an initial build system for the project based on [Webpack](https://webpack.js.org/) and [Django-webpack-loader](https://github.com/ezhome/django-webpack-loader). Webpack bundles up our static assets and outputs them into `public/static/bundles`, while Django-webpack-loader lets us render the right bundle in our templates.

**More details**
* Compiles Scss and ES6 and JSX syntax
* Watches for any changes, recompiles bundle and livereloads the page
* JS bundle gets outputted to `public/static/bundles`, old bundles get removed
* Scss gets compiled and inlined in the template
* Source maps are enabled 🙌
* Sets up React and Foundation

### How To Run/Test
- Use `nvm install` to make sure you have the right version of Node (7.10.0)
- `pip install -r requirements.txt` to install Django-webpack-loader
- `npm install` for all the frontend things
- `./manage.py runserver`
- `npm run start` to build the bundle and start watching for changes
- Open up a page like http://localhost:8000/public/ia/legislators/ocd-person/c918e4fa-d500-4f2e-a29c-5e22a35292ee
- Now, make some changes to `public/static/js/index.js` or `public/static/styles/app.scss`. Check the page— it should have gotten reloaded and updated with your compiled changes.


### Not in this PR
This PR doesn't include a production-ready configuration. which we should do later on. That work should include:
- Extracting inlined styles to a CSS file to enable minifying and caching
- Minifying and gzipping CSS and JS files
- Disabling injected livereload script on prod
- Disable sourcemaps on prod
- Autoprefixing CSS/JS for browser support
